samtranslator/model/api/api_generator.py

```diff
@@ -697,6 +697,9 @@ def _construct_usage_plan(self, rest_api_stage=None):
         if auth_properties.UsagePlan is None:
             return []
         usage_plan_properties = auth_properties.UsagePlan
+        # throws error if UsagePlan is not a dict
+        if not isinstance(usage_plan_properties, dict):
+            raise InvalidResourceException(self.logical_id, "Invalid type for field 'UsagePlan'")
```

Comment:
(minor) it might be more user-friendly to tell them the expected type `"UsagePlan" must be a dictionary` rather than just saying it's invalid.

Reply:
good one, updated